pkg/kubectl/cmd/util/factory.go

```diff
@@ -74,90 +74,108 @@ const (
 	// PauseObject marks the provided object as paused ie. it will not be reconciled by its controller.
-	PauseObject func(object runtime.Object) (bool, error)
+	PauseObject(object runtime.Object) (bool, error)
 	// ResumeObject resumes a paused object ie. it will be reconciled by its controller.
-	ResumeObject func(object runtime.Object) (bool, error)
+	ResumeObject(object runtime.Object) (bool, error)
 	// Returns a schema that can validate objects stored on disk.
-	Validator func(validate bool, cacheDir string) (validation.Schema, error)
+	Validator(validate bool, cacheDir string) (validation.Schema, error)
 	// SwaggerSchema returns the schema declaration for the provided group version kind.
-	SwaggerSchema func(unversioned.GroupVersionKind) (*swagger.ApiDeclaration, error)
+	SwaggerSchema(unversioned.GroupVersionKind) (*swagger.ApiDeclaration, error)
 	// Returns the default namespace to use in cases where no
 	// other namespace is specified and whether the namespace was
 	// overridden.
-	DefaultNamespace func() (string, bool, error)
+	DefaultNamespace() (string, bool, error)
 	// Generators returns the generators for the provided command
-	Generators func(cmdName string) map[string]kubectl.Generator
+	Generators(cmdName string) map[string]kubectl.Generator
 	// Check whether the kind of resources could be exposed
-	CanBeExposed func(kind unversioned.GroupKind) error
+	CanBeExposed(kind unversioned.GroupKind) error
 	// Check whether the kind of resources could be autoscaled
-	CanBeAutoscaled func(kind unversioned.GroupKind) error
+	CanBeAutoscaled(kind unversioned.GroupKind) error
 	// AttachablePodForObject returns the pod to which to attach given an object.
-	AttachablePodForObject func(object runtime.Object) (*api.Pod, error)
+	AttachablePodForObject(object runtime.Object) (*api.Pod, error)
 	// UpdatePodSpecForObject will call the provided function on the pod spec this object supports,
 	// return false if no pod spec is supported, or return an error.
-	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+	UpdatePodSpecForObject(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
 	// EditorEnvs returns a group of environment variables that the edit command
 	// can range over in order to determine if the user has specified an editor
 	// of their choice.
-	EditorEnvs func() []string
+	EditorEnvs() []string
 	// PrintObjectSpecificMessage prints object-specific messages on the provided writer
-	PrintObjectSpecificMessage func(obj runtime.Object, out io.Writer)
+	PrintObjectSpecificMessage(obj runtime.Object, out io.Writer)
+	// Command will stringify and return all environment arguments ie. a command run by a client
+	// using the factory.
+	// TODO: We need to filter out stuff like secrets.
+	Command() string
+	// BindFlags adds any flags that are common to all kubectl sub commands.
+	BindFlags(flags *pflag.FlagSet)
```

Comment:
If these weren't exposed before, lets keep them hidden.  I don't think we want to encourage access patterns like this.